vdr/didx509/resolver_test.go

```diff
@@ -0,0 +1,121 @@
+package didx509
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/minio/sha256-simd"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/pki"
+	resolver2 "github.com/nuts-foundation/nuts-node/vdr/resolver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"testing"
+)
+
+func TestManager_Resolve(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	validator := pki.NewMockValidator(ctrl)
+	resolver := NewResolver(validator)
+	metadata := resolver2.ResolveMetadata{}
+
+	otherNameValue := "A_BIG_STRING"
+	_, certChain, rootCertificate, _, signingCert, err := BuildCertChain(otherNameValue)
+	require.NoError(t, err)
+	metadata.X509CertChain = certChain
+	metadata.X509CertThumbprint = sha1Sum(signingCert.Raw)
+	metadata.X509CertThumbprintS256 = sha256Sum(signingCert.Raw)
+
+	rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "sha256", sha256Sum(rootCertificate.Raw), otherNameValue))
+
+	t.Run("test nulls", func(t *testing.T) {
+		chain := metadata.X509CertChain
+		metadata.X509CertChain = nil
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), errors.New("x509 rootCert chain is missing").Error())
+		metadata.X509CertChain = chain
+
+	})
+	t.Run("happy flow", func(t *testing.T) {
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
+		resolve, documentMetadata, err := resolver.Resolve(rootDID, &metadata)
+		require.NoError(t, err)
+		assert.NotNil(t, resolve)
+		assert.NotNil(t, documentMetadata)
+	})
+	t.Run("happy flow 2", func(t *testing.T) {
```

Comment:
"2" isn't really informative

Reply:
Updated all test names to more semantic names